kivy/core/text/__init__.py

```diff
@@ -659,14 +660,15 @@ def render_lines(self, lines, options, render_text, y, size):
                 line = last_word.text
                 if not cur_base_dir:
                     cur_base_dir = find_base_dir(line)
-            x = xpad
+            x = padding_left
             if halign == 'auto':
                 if cur_base_dir and 'rtl' in cur_base_dir:
-                    x = max(0, int(w - lw - xpad))  # right-align RTL text
+                    # right-align RTL text
+                    x = max(0, int(w - lw - padding_right))
             elif halign == 'center':
                 x = int((w - lw) / 2.)
```

Comment:
Since `padding_left` and `padding_right` now can be different, we should also center the text appropriately? (by also taking into account `rtl` text)

Reply:
Previously, only the right padding (the same as the left padding) was taken into account, so I think it's ok, or not?

Reply:
I'm referring to the `halign == 'center'` case.

Previously, with only 1 `padding_x` padding value, the padding was in-influent to calculate the text positioning.
Now, instead, with `padding_left` and `padding_right`, we should take it into account.